packages/modules/index/src/services/postgres-provider.ts

```diff
@@ -353,6 +227,91 @@ export class PostgresProvider {
     }
   }
 
+  @InjectManager("baseRepository_")
+  async query<const TEntry extends string>(
+    config: IndexTypes.IndexQueryConfig<TEntry>,
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<IndexTypes.QueryResultSet<TEntry>> {
+    await this.#isReady_
+
+    const {
+      keepFilteredEntities,
+      fields = [],
+      filters = {},
+      joinFilters = {},
+    } = config
+    const { take, skip, order: inputOrderBy = {} } = config.pagination ?? {}
+
+    const select = normalizeFieldsSelection(fields)
+    const where = flattenObjectKeys(filters)
+    const joinWhere = flattenObjectKeys(joinFilters)
+    const orderBy = flattenObjectKeys(inputOrderBy)
+
+    const { manager } = sharedContext as { manager: SqlEntityManager }
+    let hasPagination = false
+    if (typeof take === "number" || typeof skip === "number") {
```

Comment:
I believe we can have a `take` that is not a pagination.
I would only consider to paginate if we have `skip`.